doc/proxy-metrics.md

```diff
@@ -0,0 +1,71 @@
++++
+title = "Proxy Metrics"
+docpage = true
+[menu.docs]
+  parent = "proxy-metrics"
++++
+
+The Conduit proxy exposes metrics that describe the traffic flowing through the
+proxy.  These metrics are exposed in Prometheus format on the proxy control port
+(4191 by default) at the `/metrics` path.  The following metrics will be
+exposed:
+
+### `request_total`
+
+A counter of the number of requests the proxy has received.  This is incremented
+when the request stream begins.
+
+### `request_duration_ms`
+
+A histogram of the duration of a request.  This is measured from when the
+request headers are received to when the request stream has completed.
+
+### `response_total`
+
+A counter of the number of responses the proxy has received.  This is
+incremeneted when the response stream ends.
+
+### `response_duration_ms`
+
+A histogram of the duration of a response.  This is measured from when the
+response headers are received to when the response stream has completed.
+
+### `response_latency_ms`
+
+A histogram of the total latency of a response.  This is measured from when the
+request headers are received to when the response stream has completed.
+
+## Labels
+
+Each of these metrics has the following labels:
+
+* `direction`: `inbound` if the request originated from outside of the pod,
+               `outbound` if the request originated from inside of the pod.
+* `deployment`: The deployment that the pod belongs to (if applicable).
+* `job`: The job that the pod belongs to (if applicable).
+* `replica_set`: The replica set that the pod belongs to (if applicable).
+* `daemon_set`: The daemon set that the pod belongs to (if applicable).
+* `replication_controller`: The replication controller that the pod belongs to 
+                            (if applicable).
+* `authority`: The value of the `:authority` (HTTP/2) or `Host` (HTTP/1.1)
+               header of the request.
+* `dst_deployment`: The deployment to which this request is being sent.  Only
+                    applicable if `direction=outbound`.
+* `dst_job`: The job to which this request is being sent.  Only applicable if
+             `direction=outbound`.
+* `dst_replica_set`: The replica set to which this request is being sent.  Only
+                     applicable if `direction=outbound`.
+* `dst_daemon_set`: The daemon set to which this request is being sent.  Only
+                    applicable if `direction=outbound`.
+* `dst_replication_controller`: The replication controller to which this request
+                                is being sent.  Only applicable if
+                                `direction=outbound`.
+* `dst_namespace`: The namespace to which this request is being sent.  Only
+                   applicable if `direction=outbound`.
```

Comment:
same wrt these -- `dst_k8s_ns`, etc

though i think we should probably explicitly separate k8s labels from proxy-defined labels in this documentation.

Reply:
Similarly, these labels come from the destination service so the proxy is agnostic to them.  Easy to update them in the destination service if desired.  For the moment, I'm not sure separating proxy and k8s labels makes sense.  In the future if the Conduit proxy is used in non-k8s environments then yes, we certainly should split these up.